File: agent/internal/types/messages.go
package types

import "time"

type AgentMetricsSummary struct {
	CPUPct   float64 `json:"cpu_pct"`
	MemBytes uint64  `json:"mem_bytes"`
}

type ContainerResourceSample struct {
	ID            string  `json:"id"`
	Name          string  `json:"name"`
	CPUPct        float64 `json:"cpu_pct"`
	MemBytes      uint64  `json:"mem_bytes"`
	MemLimitBytes uint64  `json:"mem_limit_bytes"`
	NetIOBytes    uint64  `json:"net_io_bytes"`
}

type ContainerStatsBatch struct {
	Type         string                    `json:"type"`
	AgentID      string                    `json:"agent_id"`
	AgentLabel   string                    `json:"agent_label,omitempty"`
	SentAt       time.Time                 `json:"sent_at"`
	Sequence     uint64                    `json:"sequence"`
	Containers   []ContainerResourceSample `json:"containers"`
	AgentMetrics AgentMetricsSummary       `json:"agent_metrics"`
}

type AgentStatusMessage struct {
	Type       string    `json:"type"`
	AgentID    string    `json:"agent_id"`
	AgentLabel string    `json:"agent_label,omitempty"`
	SentAt     time.Time `json:"sent_at"`
	UptimeSecs uint64    `json:"uptime_secs"`
	Version    string    `json:"version,omitempty"`
	Features   []string  `json:"features,omitempty"`
}

type DashboardMessage interface{}

File: web/src/lib/types/messages.ts
export type AgentConnectionState = 'placeholder' | 'connecting' | 'connected' | 'error';

export interface AgentEndpoint {
	id: string;
	url: string;
	label: string;
	status: AgentConnectionState;
	lastSeenAt: string | null;
	createdAt: string;
	notes?: string;
}

export interface AgentMetricsSummary {
        cpu_pct: number;
        mem_bytes: number;
}

export interface AgentCpuSample {
        at: string;
        cpu_pct: number;
}

export interface ContainerResourceSample {
        id: string;
        name: string;
        cpu_pct: number;
        mem_bytes: number;
	mem_limit_bytes: number;
	net_io_bytes: number;
}

export interface ContainerStatsBatch {
	type: 'container_stats_batch';
	agent_id: string;
	agent_label?: string;
	sent_at: string;
	sequence: number;
	containers: ContainerResourceSample[];
	agent_metrics: AgentMetricsSummary;
}

export interface AgentStatusMessage {
	type: 'agent_status';
	agent_id: string;
	agent_label?: string;
	sent_at: string;
	uptime_secs: number;
	version?: string;
	features?: string[];
}

export type DashboardMessage = ContainerStatsBatch | AgentStatusMessage;
